code/modules/games/cards.dm

```diff
@@ -0,0 +1,435 @@
+	cards -= P
+	H.update_icon()
+	user.visible_message("\The [user] draws a card.")
+	to_chat(user,"It's the [P].")
+
+/obj/item/weapon/deck/verb/deal_card()
+
+	set category = "Object"
+	set name = "Deal"
+	set desc = "Deal a card from a deck."
+	set src in view(1)
+
+	if(usr.stat || !Adjacent(usr)) return
+
+	if(!cards.len)
+		to_chat(usr,"There are no cards in the deck.")
+		return
+
+	var/list/players = list()
+	for(var/mob/living/player in viewers(3))
+		if(!player.stat)
+			players += player
+	//players -= usr
+
+	var/mob/living/M = input("Who do you wish to deal a card?") as null|anything in players
+	if(!usr || !src || !M) return
+
+	deal_at(usr, M, 1)
+
+/obj/item/weapon/deck/verb/deal_card_multi()
+
+	set category = "Object"
+	set name = "Deal Multiple Cards"
+	set desc = "Deal multiple cards from a deck."
+	set src in view(1)
+
+	if(usr.stat || !Adjacent(usr)) return
+
+	if(!cards.len)
+		to_chat(usr,"There are no cards in the deck.")
+		return
+
+	var/list/players = list()
+	for(var/mob/living/player in viewers(3))
+		if(!player.stat)
+			players += player
+	//players -= usr
+	var/maxcards = max(min(cards.len,10),1)
+	var/dcard = input("How many card(s) do you wish to deal? You may deal up to [maxcards] cards.") as num
+	if(dcard > maxcards)
+		return
+	var/mob/living/M = input("Who do you wish to deal [dcard] card(s)?") as null|anything in players
+	if(!usr || !src || !M) return
+
+	deal_at(usr, M, dcard)
+
+/obj/item/weapon/deck/proc/deal_at(mob/user, mob/target, dcard) // Take in the no. of card to be dealt
+	var/obj/item/weapon/hand/H = new(get_step(user, user.dir))
+	var/i
+	for(i = 0, i < dcard, i++)
+		H.cards += cards[1]
+		cards -= cards[1]
+		H.concealed = 1
+		H.update_icon()
+	if(user==target)
+		user.visible_message("\The [user] deals [dcard] card(s) to \himself.")
+	else
+		user.visible_message("\The [user] deals [dcard] card(s) to \the [target].")
+	H.throw_at(get_step(target,target.dir),10,1,H)
+
+
+/obj/item/weapon/hand/attackby(obj/O as obj, mob/user as mob)
+	if(cards.len == 1 && istype(O, /obj/item/weapon/pen))
+		var/datum/playingcard/P = cards[1]
+		if(P.name != "Blank Card")
+			to_chat(user,"You cannot write on that card.")
+			return
+		var/cardtext = sanitize(input(user, "What do you wish to write on the card?", "Card Editing") as text|null, MAX_PAPER_MESSAGE_LEN)
+		if(!cardtext)
+			return
+		P.name = cardtext
+		// SNOWFLAKE FOR CAG, REMOVE IF OTHER CARDS ARE ADDED THAT USE THIS.
+		P.card_icon = "cag_white_card"
+		update_icon()
+	else if(istype(O,/obj/item/weapon/hand))
+		var/obj/item/weapon/hand/H = O
+		for(var/datum/playingcard/P in cards)
+			H.cards += P
+		H.concealed = src.concealed
+		qdel(src)
+		H.update_icon()
+		return
+	..()
+
+/obj/item/weapon/deck/attack_self()
+	shuffle()
+
+/obj/item/weapon/deck/verb/verb_shuffle()
+	set category = "Object"
+	set name = "Shuffle"
+	set desc = "Shuffle the cards in the deck."
+	set src in view(1)
+	shuffle()
+
+/obj/item/weapon/deck/proc/shuffle()
```

Comment:
I'm sure there's already a helper for shuffling a list which could be used in this proc.